qiskit/mapper/_coupling.py

```diff
@@ -68,7 +68,7 @@ class Coupling:
 
     # pylint: disable=invalid-name
 
-    def __init__(self, couplingdict=None):
+    def __init__(self, couplingdict=None, couplinglist=None, regname='q'):
```

Comment:
as we are moving away from and making the couplingdict deprecated can we make this just take in the couplinglist

Reply:
If we keep i would give a warning then call the converter